src/app/services/comment-period.service.ts

```diff
@@ -0,0 +1,114 @@
+import { Injectable } from '@angular/core';
+import { Response } from '@angular/http';
+import { CommentPeriod } from '../models/commentperiod';
+import { Comment } from '../models/comment';
+import { Document } from '../models/document';
+import { Project } from '../models/project';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/switchMap';
+
+import { Api } from './api';
+import { ValuedComponent } from '../models/vcs';
+
+@Injectable()
+export class CommentPeriodService {
+  pcp: CommentPeriod;
+
+  constructor(private api: Api) { }
+
+  // return a public comment period object
+  getByCode(id: string, code: string): Observable<CommentPeriod> {
+  // Grab the project data first
+    return this.api.getPCPByCode(id)
+      .map((res: Response) => res.json())
+      .map((pcp: CommentPeriod) => {
```

Comment:
Please change type to `any`. This is a JSON response coming from the API, not client side CommentPeriod object (although they are quite similar)